src/sonic-host-services/scripts/hostcfgd

```diff
@@ -966,6 +970,67 @@ class NtpCfg(object):
             syslog.syslog(syslog.LOG_INFO, 'ntp server update, restarting ntp-config, ntp servers configured {}'.format(self.ntp_servers))
             run_cmd(cmd)
 
+class PamLimitsCfg(object):
+    """
+    PamLimit Config Daemon
+    1) The pam_limits PAM module sets limits on the system resources that can be obtained in a user-session.
+    2) Purpose of this daemon is to render pam_limits config file.
+    """
+    def __init__(self, CfgDb):
+        self.config_db = CfgDb
+        self.hwsku = ""
+        self.type = ""
+
+    # Load config from ConfigDb and render config file/
+    def update_config_file(self):
+        device_metadata = self.config_db.get_table('DEVICE_METADATA')
+        if "localhost" not in device_metadata:
+            return
+
+        self.read_localhost_config(device_metadata["localhost"])
+        self.render_conf_file()
+
+    # Read localhost config
+    def read_localhost_config(self, localhost):
+        if "hwsku" in localhost:
```

Comment:
>in [](http://example.com/codeflow?start=19&length=2)

We need to initialize `self.hwsku` and `self.type` if expected data is missing in ConfigDB.

`read_localhost_config()` could be called multiple times. If not initialized, it will keep whatever old values.

Reply:
Fixed, reset these 2 members to empty when they are not exist in config DB.